tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioHelperTests.cs

```diff
@@ -257,5 +258,35 @@ public void ValidateRequest_Should_Fail_With_NonMatching_Signature()
                 TwilioHelper.RequestToActivity(httpRequest.Object, string.Empty, string.Empty);
             });
         }
+
+        [Fact]
+        public void Test()
+        {
+            // Setup mocked ITwilioAdapterOptions
+            var options = new Mock<ITwilioAdapterOptions>().SetupAllProperties();
+            options.Object.AccountSid = "MockTest";
+            options.Object.AuthToken = "MockToken";
+            options.Object.TwilioNumber = "MockNumber";
+
+            // Setup mocked Twilio API client
+            const string resourceIdentifier = "Mocked Resource Identifier";
+            var twilioApi = new Mock<ITwilioClient>();
+            twilioApi.Setup(x => x.GetResourceIdentifier(It.IsAny<object>())).Returns(Task.FromResult(resourceIdentifier));
+
+            // Create a new Twilio Adapter with the mocked classes and get the responses
+            var twilioAdapter = new TwilioAdapter(options.Object, twilioApi.Object);
```

Comment:
_Future refactor_: extract the creation of this _TwilioAdapter_ (including the _options_ and the _twilioApi_) to another method.

Reply:
I heartily agree. This change would greatly improve the readability of the code.